x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/rollup/action/RollupShardTask.java

```diff
@@ -9,15 +9,18 @@
 
 import org.elasticsearch.action.downsample.DownsampleConfig;
 import org.elasticsearch.index.shard.ShardId;
-import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.persistent.AllocatedPersistentTask;
+import org.elasticsearch.persistent.PersistentTasksService;
 import org.elasticsearch.tasks.TaskId;
+import org.elasticsearch.tasks.TaskManager;
 import org.elasticsearch.xpack.core.rollup.RollupField;
 
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
 
-public class RollupShardTask extends CancellableTask {
+public class RollupShardTask extends AllocatedPersistentTask {
+    public static final String TASK_NAME = "rollup-shard";
```

Comment:
Can we rename this to `downsampel-shard`?

Reply:
Everything in the code is called "Rollup"<something>...can we, maybe, do this in a separate PR were we rename everything? If I do it here this PR will grow very large.